orderBook.js

```diff
@@ -0,0 +1,43 @@
+const reconcileOrder = (existingBook, incomingOrder) =>
+  existingBook.length
+    ? checkAllTypesMatch(existingBook, incomingOrder)
+    : addToBook(existingBook, incomingOrder)
+
+const checkAllTypesMatch = (existing, incoming) =>
```

Comment:
This isn't checking if types match it is actually performing operations based on the types. Naming becomes difficult because the function is doing multiple things instead of one. The real function name would need to be `addToBookIfTypesMatchElseCheckPriceMatches` which is a clear tip off that this function isn't written well.